kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -640,7 +640,7 @@ extension [A, E, Ctx](effect: A < (Abort[E] & Async & Ctx))
     def fork(
         using
         flat: Flat[A],
-        boundary: Boundary[Ctx, IO],
+        boundary: Boundary[Ctx, IO & Abort[E]],
```

Comment:
The second type parameter of `Boundary` represents effects that are allowed in addition to `ContextEffect`s. `Abort` was missing, which is the main cause of the bug.